### ISSUE TITLE
[BUG] Supported `kubernetes` library is severely outdated

### ISSUE THREAD

**Issue by lkubb** [salt-extensions/saltext-kubernetes#5]
**Description**
This extension only supports severely outdated versions of the `kubernetes` library. For details, see https://github.com/saltstack/salt/issues/62621 and https://github.com/saltstack/salt/issues/63090

**Setup**
irrelevant

**Steps to Reproduce the behavior**
install the extension

**Expected behavior**
Support recent versions

**Screenshots**
We're currently pinning `kubernetes==3.0.0` from 2017. This causes some installation issues in the Windows/macOS test jobs.

**Versions Report**
1.0.1

**Comment by benfiedler** [comment]
Agreed - I have made and tested some offline changes to the state+module (which I hope to eventually contribute back), but I bumped the kubernetes module version to [19.15.0](https://pypi.org/project/kubernetes/19.15.0/). Reason I picked that one is it's the first version that lists supports for python 3.10.

I think the bare minimum changes needed for `kubernetesmod.py` to function with 19.15.0 was to remove 'beta' from the hard coded entries for deployment/deployment spec, e.g. `AppsV1beta1DeploymentSpec` -> `AppsV1DeploymentSpec`
